Docs 5941 mmp cloudhub v4.2 (#1014)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/mmp-concept.adoc

> = Mule Maven Plugin Reference

Reviewer: This isn't a reference, which implies data, in a table or not, that is primarily not narrative.
It looks to me like this is just the overview/about topic for the plugin. Maybe "What the Mule Maven Plugin Is" or "Goals of Using the Mule Maven Plugin" or "What You Can Do Using the Mule Maven Plugin" or similar?

ifndef::env-site,env-github[]
include::_attributes.adoc[]

> endif::[]

Reviewer: HIDDEN TEXT **edit**:

The Mule Maven plugin enables you to integrate the packaging, testing, and deployment of your Mule applications with your Maven lifecycle. +
**This is a good intro sentence. 
However, I see that this list of three uses does not precisely match the three goals that you describe later: should it?
I think that this sentence should be modified to match the three goals, and that the sentences about each goal should be moved to their subtopics, and the xref moved to the packaging subtopic.
After the intro sentence, I would then use the sentence about compatibility and not make that separate.**
See xref:package-a-mule-application.adoc[Package a Mule Application] for packaging instructions.
**move this to packaging goal subtopic**
With the plugin, you can deploy your Mule application to different deployment targets, such as standalone Mule instances, clustered instances, and CloudHub.
**move this to the deployment goal subtopic**
The Mule Maven plugin is compatible with Mule runtime engine (Enterprise Edition) and with Mule Kernel (Community Edition).
**I don't think that these names are current (at least, not Kernel). I suggest replacing this with:
"The Mule Maven plugin is compatible with Mule 4.1 and Anypoint Studio 7.x."**
== Compatibility
* Mule 4.x
* Anypoint Studio 7.x
**omit**
== Mule Maven Plugin Goals
**Replace with three subtopics at this level, one for each of Packaging, Testing, and Deployment...although I don't actually see Testing below but I do see Undeploy, which is really part of Deployment.**

Each goal accepts parameters that are unique to the desired plugin behavior.
To provide a parameter from the command line, prepend `-D` to the parameter name.

> 

Reviewer: See edit comments above.
And save the `-D` syntax requirement for text that is actually about each set of parameters?

[[package-goal]]

> === Package Goal

Reviewer: See prev. suggestions for reorg.
Suggest that we remove "goal" from titles so that someone skimming content does not wonder what that refers to.

Reviewer: I am going to just copyedit the rest, because I am behind schedule.

> The `package` goal generates the application JAR file. This goal binds by default to the http://maven.apache.org/ref/3.6.1/maven-core/lifecycles.html[Maven lifecycle phase]: `package`.

Reviewer: ```suggestion
The packaging task generates the application JAR file, binding by default to the http://maven.apache.org/ref/3.6.1/maven-core/lifecycles.html[Maven lifecycle phase]: `package`.
```
Binding to a lifecycle phase? Binding what? to a phase? or are you saying that the phase is called `package`? Clarify.

> Optional Parameters

Reviewer: Optional parameters to what command? What is the context?

[%header%autowidth.spread,cols="a,a,a"]
|===
|Name | Type | Description

> |`onlyMuleSources` | boolean | Generates the application JAR file containing only the source code. This property is just for sharing purposes.

Reviewer: ```suggestion
|`onlyMuleSources` | boolean | Generates the application JAR file containing only the source code, so that it can be shared.
```

|`attachMuleSources` | boolean | Attaches the source code inside the generated JAR file.

> |`lightweightPackage` | boolean | Doesn't generate the repository with all the application dependencies inside the JAR file. This property is just for sharing purposes.

Reviewer: ```suggestion
|`lightweightPackage` | boolean | Doesn't generate the repository with all the application dependencies inside the JAR file, so that the file can be shared.
```

|===

> For example, to execute the `package` goal and set the `attachMuleSources` parameter, you run the following command:

Reviewer: ```suggestion
For example, to execute packaging that includes a repository without including application dependencies, run the following command:
```

[source,console,linenums]
----
mvn package -DattachMuleSources
----

=== Deploy Goal

> This goal uploads and deploys the application JAR file to any of the application deployment targets. The deploy goal binds by default to the http://maven.apache.org/ref/3.6.1/maven-core/lifecycles.html[Maven lifecycle phase]: `deploy`.

Reviewer: ditto comments for the previous topic

> 

Reviewer: HIDDEN TEXT unedited

> By default, this plugin allows you to deploy using five different deployment strategies:

Reviewer: ```suggestion
By default, this plugin enables you to deploy your Mule app by using any of five different deployment strategies:
```

* CloudHub deployment +

> Automate the deployment of your Mule Application to CloudHub. See xref:deploy-to-cloudhub.adoc[Deploy to CloudHub] for instructions.

Reviewer: ```suggestion
Automates the deployment of your Mule application to CloudHub. See xref:deploy-to-cloudhub.adoc[Deploy to CloudHub] for instructions.
```

* Runtime Fabric deployment. +

> Deploy your Mule Application to Runtime Fabric.

Reviewer: ```suggestion
Deploys your Mule application to Anypoint Runtime Fabric.
```

Reviewer: Ditto previous edits in the HIDDEN TEXT that follows, about the other three methods.

> When deploying an application using the Mule Maven plugin you need to specify a valid deployment configuration in your `pom.xml` file. +

Reviewer: ```suggestion
When deploying an application using the Mule Maven plugin, you must specify a valid deployment configuration in your `pom.xml` file. +
```

> You can set up a deployment configuration using their dedicated deployment references.

Reviewer: Who is "they" of "their dedicated deployment references"? Are we referring to documentation other than our own here? Clarify.

> == Deploy a Mule Application to Runtime Fabric

Reviewer: Avoid jumping from one title directly to another. Try to provide a useful introduction to the task, clarifying in what context, under what circumstances, the audience wants or needs to perform this task, perhaps instead of another task.
